### ISSUE TITLE
SearchStream RESOURCE_EXHAUSTED

### ISSUE THREAD

**Issue by duferdev** [googleads/google-ads-java#583]
**Describe the bug:**

Hi there,

We have migrated from API V9 (ads-lib 16.0.3) to V10 (ads-lib 17.0.0) 

Since then we start having report request issues in multiple accounts like:

17:55:17.527 [grpc-default-executor-0hread] com.google.ads.googleads.lib.logging.RequestLogger.logAtLevel(RequestLogger.java:157)evel com.google.ads.googleads.lib.logging.RequestLogger.logAtLevel(RequestLogger.java:157)ogger - FAILURE REQUEST DETAIL.

Request
-------
MethodName: google.ads.googleads.v10.services.GoogleAdsService/SearchStream
Endpoint: [googleads.googleapis.com:443](http://googleads.googleapis.com:443/)
Headers: {developer-token=REDACTED, login-customer-id=XXXXX, x-goog-api-client=gl-java/1.8.0_301 gccl/17.0.0 gapic/17.0.0 gax/2.6.1 grpc/1.41.0}
Body: customer_id: "XXXXXX"
query: "SELECT segments.date,segments.device,[campaign.id](http://campaign.id/),[campaign.name](http://campaign.name/),campaign.status,[ad_group.id](http://ad_group.id/),[ad_group.name](http://ad_group.name/),ad_group.status,ad_group_criterion.criterion_id,ad_group_criterion.keyword.text,ad_group_criterion.status,ad_group_criterion.labels,ad_group_criterion.keyword.match_type,ad_group_criterion.final_urls,ad_group_criterion.effective_cpc_bid_micros,ad_group_criterion.quality_info.quality_score,metrics.impressions,metrics.clicks,metrics.cost_micros,metrics.ctr,metrics.conversions,metrics.average_cpc FROM keyword_view WHERE ad_group_criterion.negative = false AND ad_group_criterion.criterion_id != 3000000 AND metrics.impressions > 0 AND segments.date BETWEEN \"2022-02-28\" AND \"2022-03-01\""

Response
--------
Headers: Metadata(content-disposition=attachment,content-type=application/grpc,date=Wed, 02 Mar 2022 16:55:17 GMT,alt-svc=h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43")
Body: null
Failure message: null
Status: Status{code=RESOURCE_EXHAUSTED, description=gRPC message exceeds maximum size 4194304: 5529741, cause=null}.sg

Any thoughts?

Thanks.

**Client library version and API version:**
Client library version: 17.0.0
Google Ads API version: V10
JDK version: 1.8

**Comment by jradcliff** [comment]
Hi,

This was resolved in release 17.0.1 (see #565). Please upgrade to 17.0.1 and reopen this issue if you still encounter errors.

Thanks,
Josh

**Comment by PeterLavetsky** [comment]
Hi ... we're getting intermittent days of high frequency resource exhausted errors ... no new code was released in between examples of low frequency days and high frequency days. We're on 38.0.0 of the Java client library for Google Ads API

Thanks
Pete

```
 ./resource-exhausted-test.sh 
/efs/api/google/requests/new-google-requests-summary.2025-06-28.log 73903
/efs/api/google/requests/new-google-requests-summary.2025-06-27.log 2
/efs/api/google/requests/new-google-requests-summary.2025-06-26.log 16
/efs/api/google/requests/new-google-requests-summary.2025-06-25.log 2
/efs/api/google/requests/new-google-requests-summary.2025-06-24.log 4
/efs/api/google/requests/new-google-requests-summary.2025-06-23.log 11
/efs/api/google/requests/new-google-requests-summary.2025-06-22.log 29519
/efs/api/google/requests/new-google-requests-summary.2025-06-21.log 84703
/efs/api/google/requests/new-google-requests-summary.2025-06-20.log 5
/efs/api/google/requests/new-google-requests-summary.2025-06-19.log 1
/efs/api/google/requests/new-google-requests-summary.2025-06-18.log 6
/efs/api/google/requests/new-google-requests-summary.2025-06-17.log 1
/efs/api/google/requests/new-google-requests-summary.2025-06-16.log 6
/efs/api/google/requests/new-google-requests-summary.2025-06-15.log 1
/efs/api/google/requests/new-google-requests-summary.2025-06-14.log 4
/efs/api/google/requests/new-google-requests-summary.2025-06-13.log 3
/efs/api/google/requests/new-google-requests-summary.2025-06-12.log 3
/efs/api/google/requests/new-google-requests-summary.2025-06-11.log 2
/efs/api/google/requests/new-google-requests-summary.2025-06-10.log 3
/efs/api/google/requests/new-google-requests-summary.2025-06-09.log 5
/efs/api/google/requests/new-google-requests-summary.2025-06-08.log 3
```
